### ISSUE TITLE
[Death Knight] Talent Bloodworms

### ISSUE THREAD

**Issue by Nevahs** [dalaranwow/dalaran-wow#2095]
As per issue #531 Bloodworms are still aggressive and attack nearby mobs within aggro range, even without the player or the mob having engaged each other.

It is as if they were a pet, set to aggressive, while they should be set to defensive.

**Comment by taralej90** [comment]
Will test this issue these days and provide feedback. Blood worms should act as GUARDIANs, attacking only targets which have threat generated against the DK.

**Comment by taralej90** [comment]
Feedback after testing: Bloodworms are dubbed and act as Guardians. 2-4 spawn, immediately attacking any target on which the DK has generated threat, either through pulling and dealing damage or by being too close to Hostile NPCs. They do not attack targets on which the DK doesn't have threat - be it Hostile targets which haven't engaged the DK or Neutral targets, which was the biggest issue before.

As a clarification, the act of pulling by being in range of a Hostile target generates threat by itself.

tl;dr In conclusion, Bloodworms work perfectly. Whether they damage and heal correctly I do not know, but the numbers seem OK to me.

**Comment by Rushor** [comment]
Scripts/Pet: DK - Bloodworm - no longer attack hostile creatures in range by default

* closes https://github.com/dalaranwow/dalaran-wow/issues/2095
* commit: b3f30c993e377f6df9f717d4a1a22d9d0de92038

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by TeridanDW** [comment]
Works as intended on PTR (worms now have defensive stance).

**Comment by AttendantTock** [comment]
Hello,

The bloodworms seems to work against the dummies on the PTR, but **not** in duels.

See unlisted video:
https://youtu.be/v_iTzedwZxE

**Comment by thanhtotti** [comment]
Still bugged for me in live server.
